src/awkward/_connect/cuda/cuda_kernels/awkward_ListArray_min_range.cu

```diff
@@ -0,0 +1,21 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+template <typename T, typename C, typename U>
+__global__ void
+awkward_ListArray_min_range(T* tomin,
+                            const C* fromstarts,
+                            const U* fromstops,
+                            int64_t lenstarts,
+                            uint64_t invocation_index,
+                            uint64_t* err_code) {
+  if (err_code[0] == NO_ERROR) {
+    int64_t thread_id = blockIdx.x * blockDim.x + threadIdx.x;
+    int64_t shorter = fromstops[0] - fromstarts[0];
+
+    if (thread_id >=1 && thread_id < lenstarts) {
+      int64_t rangeval = fromstops[thread_id] - fromstarts[thread_id];
+      shorter = (shorter < rangeval) ? shorter : rangeval;
+      atomicMin(tomin, shorter);
+    }
```

Comment:
[awkward_ListArray_min_range](https://awkward-array.org/doc/main/reference/generated/kernels.html#awkward-listarray-min-range) is a reduce-to-scalar algorithm, but here it's implemented with an atomic. Let's keep this for now (it works), but maybe add it to a list of implemented-but-not-optimal CUDA kernels. I made that list: #2987.